### DISCUSSION TITLE
simple-react-esm React code errors

### DISCUSSION THREAD

**Discussion by Jaanilj** [originjs/vite-plugin-federation#162] [Q&A]
Hi! I really like the idea of this package! So thanks for that <3

However, I noticed a simple problem in `simple-react-esm` example when using React functions. I changed the button code in [here](https://github.com/originjs/vite-plugin-federation/blob/main/packages/examples/simple-react-esm/remote-esm/src/button.js) to:

```jsx
import { useState } from 'react'

const style = {
  background: '#a3e077',
  color: '#FFFFFF',
  padding: 12
}

export default Button = () => {
  const [count, setCount] = useState(0)

  const buttonClick = () => setCount((c) => c + 1)

  return <div style={style}>
    <span>Current count {count}</span>
    <button onClick={buttonClick}>Increment count</button>
  </div>
}
```

So I wanted to use `useState` (Shared from `host-esm`).

To my surprise, I got an error that `Button` is not defined. Looking at the built code, I see a weird build result that is not valid JS.

<img width="604" alt="image" src="https://user-images.githubusercontent.com/14058634/161315921-50fd8e74-04f8-4721-9ceb-5ff731d6c16c.png">

I fixed this and got a new error:

<img width="1102" alt="image" src="https://user-images.githubusercontent.com/14058634/161316093-27c15a5a-3231-4f69-8023-15062bab3df9.png">

I think the React share doesn't work in the example...?

**Comment by Jiannan-dev** [comment]
@Jaanilj  Hi, sorry, I've been a bit busy lately, so I'm just getting back to you now. I tried to run it using your code, but strangely enough, in my webstorm, this usage prompts an error, like this
![image](https://user-images.githubusercontent.com/70385062/162895882-e7f10e66-bf36-43ab-9312-7be84f8561b6.png)

Then I changed the code to the following
``` js
// i only modify here
export default function Button(){
  const [count, setCount] = useState(0)

  const buttonClick = () => setCount((c) => c + 1)

  return <div style={style}>
    <span>Current count {count}</span>
    <button onClick={buttonClick}>Increment count</button>
  </div>
}
```
Then I got the following result, and it looks like the display and buttons are working properly

![image](https://user-images.githubusercontent.com/70385062/162896739-f41233a5-d9b9-44da-a626-a9293b953ed9.png)


Is there a limitation in export default that prevents it from running?

**Reply by philals** [reply]
I am having a similar issue

Changing from 
`const [count, setCount] = useState(0);`
to
`const [count, setCount] = React.useState(0);`

Fixed it.

Will try a few other things

![image](https://user-images.githubusercontent.com/8849355/164292413-9e6662a6-14c0-4575-969a-a99388e9fa6b.png)
